src/extensions/agent_modules/shared/state_store/src/agent_state_store.c

```diff
@@ -513,251 +207,244 @@ ADUC_STATE_STORE_RESULT ADUC_StateStore_SetDeviceId(const char* deviceId)
     }
 
     sem_wait(&state_semaphore);
+
     free(state.deviceId);
     state.deviceId = SafeStrdup(deviceId);
+
     sem_post(&state_semaphore);
     return ADUC_STATE_STORE_RESULT_OK;
 }
 
 /**
- * @brief Get the DU service instance name from the state store.
+ * @brief Get the scope identifier.
+ * @return NULL if never set; else, the current scope id.
  */
-const char* ADUC_StateStore_GetDeviceUpdateServiceInstance(void)
+const char* ADUC_StateStore_GetScopeId()
 {
+    const char* value = NULL;
+
     sem_wait(&state_semaphore);
-    const char* value = state.deviceUpdateServiceInstance;
+
+    value = state.scopeId;
+
     sem_post(&state_semaphore);
     return value;
 }
 
 /**
- * @brief Set the DU service instance name in the state store.
- * @param instanceName The value to set.
+ * @brief Set the scope Id.
+ * @param scopeId The scope id.
  * @return ADUC_STATE_STORE_RESULT_OK on success, ADUC_STATE_STORE_RESULT_ERROR on failure.
  */
-ADUC_STATE_STORE_RESULT ADUC_StateStore_SetDeviceUpdateServiceInstance(const char* instanceName)
+ADUC_STATE_STORE_RESULT ADUC_StateStore_SetScopeId(const char* scopeId)
 {
+    ADUC_STATE_STORE_RESULT result = ADUC_STATE_STORE_RESULT_ERROR;
+
     sem_wait(&state_semaphore);
-    free(state.deviceUpdateServiceInstance);
-    state.deviceUpdateServiceInstance = SafeStrdup(instanceName);
+
+    free(state.scopeId);
+    state.scopeId = SafeStrdup(scopeId);
```

Comment:
maybe nit: do we know these will always be wellformed strings? Do we need to do ADUC_StrCopyN()

Reply:
Good catch. Now doing a ADUC_StrNLen check and returning ADUC_STATE_STORE_RESULT_ERROR if == 0 or == MAX_ADU_SCOPE_ID_BYTE_LEN. Also, copying with MallocAndSubstr for now since it has N bound and not as churned as ADUC_StrCopyN with later PRs (less conflicts for now).

Here is the commit that addresses this: [bound state store set of scopeId by max scopeId bytelen](https://github.com/Azure/iot-hub-device-update/pull/584/commits/720d02ee1e2ef727b0e93efa3cd56824667303f8)